app/src/src_nonfreenet/org/breezyweather/sources/mgm/MgmResultConverter.kt

```diff
@@ -0,0 +1,394 @@
+        "A" -> WeatherCode.CLEAR // Açık
+        "AB" -> WeatherCode.PARTLY_CLOUDY // Az Bulutlu
+        "PB" -> WeatherCode.PARTLY_CLOUDY // Parçalı Bulutlu
+        "CB" -> WeatherCode.CLOUDY // Çok Bulutlu
+        "HY" -> WeatherCode.RAIN // Hafif Yağmurlu
+        "Y" -> WeatherCode.RAIN // Yağmurlu
+        "KY" -> WeatherCode.RAIN // Kuvvetli Yağmurlu
+        "KKY" -> WeatherCode.SLEET // Karla Karışık Yağmurlu
+        "HKY" -> WeatherCode.SNOW // Hafif Kar Yağışlı
+        "K" -> WeatherCode.SNOW // Kar Yağışlı
+        "KYK", "YKY" -> WeatherCode.SNOW // Yoğun Kar Yağışlı
+        "HSY" -> WeatherCode.RAIN // Hafif Sağanak Yağışlı
+        "SY" -> WeatherCode.RAIN // Sağanak Yağışlı
+        "KSY" -> WeatherCode.RAIN // Kuvvetli Sağanak Yağışlı
+        "MSY" -> WeatherCode.RAIN // Mevzi Sağanak Yağışlı
+        "DY" -> WeatherCode.HAIL // Dolu
+        "GSY" -> WeatherCode.THUNDERSTORM // Gökgürültülü Sağanak Yağışlı
+        "KGY" -> WeatherCode.THUNDERSTORM // Kuvvetli Gökgürültülü Sağanak Yağışlı
+        "SIS" -> WeatherCode.FOG // Sisli
+        "PUS" -> WeatherCode.FOG // Puslu
+        "DNM" -> WeatherCode.HAZE // Dumanlı
+        "KF" -> WeatherCode.WIND // Toz veya Kum Fırtınası
+        "R" -> WeatherCode.WIND // Rüzgarlı
+        "GKR" -> WeatherCode.WIND // Güneyli Kuvvetli Rüzgar
+        "KKR" -> WeatherCode.WIND // Kuzeyli Kuvvetli Rüzgar
+        "SCK" -> null // Sıcak
+        "SGK" -> null // Soğuk
+        "HHY" -> WeatherCode.RAIN // Yağışlı
+        else -> null
+    }
+}
+
+// Simply join names of the active alert types as alert headline.
+// Turkish terminology from: https://www.mgm.gov.tr/meteouyari/turkiye.aspx?Gun=1
+// under "harita-alti-hadise"
+private fun getAlertHeadline(
+    weather: List<String>?
+): String? {
+    val items = mutableListOf<String>()
+    weather?.forEach {
+        when (it) {
+            "cold" -> items.add("Soğuk")
+            "hot" -> items.add("Sıcak")
+            "fog" -> items.add("Sis")
+            "agricultural" -> items.add("Zirai Don")
+            "ice" -> items.add("Buzlanma ve Don")
+            "dust" -> items.add("Toz Taşınımı")
+            "snowmelt" -> items.add("Kar Erimesi")
+            "avalanche" -> items.add("Çığ")
+            "snow" -> items.add("Kar")
+            "thunderstorm" -> items.add("Gökgürültülü Sağanak Yağış")
+            "wind" -> items.add("Rüzgar")
+            "rain" -> items.add("Yağmur")
+        }
+    }
+    return items.joinToString(", ")
+}
+
+// Color source: https://www.mgm.gov.tr/meteouyari/meteouyari.css
+private fun getAlertColor(
+    severity: AlertSeverity
+): Int {
+    return when (severity) {
+        AlertSeverity.EXTREME -> Color.rgb(249, 75, 101) // #btnKirmizi
+        AlertSeverity.SEVERE -> Color.rgb(255, 199, 87) // #btnTuruncu
+        AlertSeverity.MODERATE -> Color.rgb(255, 244, 49) // #btnSari
+        else -> Alert.colorFromSeverity(severity)
+    }
+}
+
+private fun getValid(
+    value: Any?
+): Any? {
```

Comment:
Why is it `Any`? You always pass a `Double` as parameter, right?

Reply:
Oh right! Lifted this piece of code from my CWA refactoring, which I had to pass both `Double` and `String`. I'll change this to `Double` only then. Will be neater above.